[iOS, macOS] Fix AvAudioSession Defaults

brminnick opened the pull request:
### Description of Change ###

This PR sets the following defaults for `AvAudioSession`:

```cs
AvAudioSession.SetCategory(AVAudioSessionCategory.PlayAndRecord, AVAudioSessionCategoryOptions.DefaultToSpeaker)
```

 ### Linked Issues ###
 <!-- Provide links to issues here (#35 will link to issue number 35). Ensure that a GitHub issue was created for your bug/proposal and it has been approved/Championed. -->

 - Fixes https://github.com/CommunityToolkit/Maui/issues/1497

 ### PR Checklist ###
 <!--
 Please check all the things you did here and double-check that you got it all, or state why you didn't do something.

 If anything is unclear please do ask :)
 -->
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)

VladislavAntonyuk commented on src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SharedSpeechToTextImplementation.macios.cs:
```diff
@@ -21,6 +21,15 @@ public sealed partial class SpeechToTextImplementation
+	public SpeechToTextImplementation()
+	{
+		var audioSession = AVAudioSession.SharedInstance();
+		audioSession.SetCategory(AVAudioSessionCategory.PlayAndRecord, AVAudioSessionCategoryOptions.DefaultToSpeaker);
```
It will be overridden in macOS (see line 32 in mac.cs file).

brminnick replied:
Good catch! What's your recommendation? 

Should we move the logic from the macOS implementation here into the shared constructor of `SpeechToTextImplementation`?

```suggestion
		if (UIDevice.CurrentDevice.CheckSystemVersion(15, 0))
{
  audioSession.SetSupportsMultichannelContent(true, out _);
}

audioSession.SetCategory(AVAudioSessionCategory.Record, AVAudioSessionCategoryOptions.DefaultToSpeaker);
```

brminnick commented:
Hey @VladislavAntonyuk! I removed the `AVAudioSession` logic from the macOS-specific implementation, and I added the `SetSupportsMultichannelContent` to the constructor as well. 

```cs
var audioSession = AVAudioSession.SharedInstance();
 if (UIDevice.CurrentDevice.CheckSystemVersion(15, 0))
 {
 	audioSession.SetSupportsMultichannelContent(true, out _);
 }

 audioSession.SetCategory(AVAudioSessionCategory.PlayAndRecord, AVAudioSessionCategoryOptions.DefaultToSpeaker);
```

If you think this looks good, let's Approve + Merge it as soon as possible in preparation for the .NET 8.0 release this week!

VladislavAntonyuk commented on src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SharedSpeechToTextImplementation.macios.cs:
```diff
@@ -21,6 +21,20 @@ public sealed partial class SpeechToTextImplementation
+			audioSession.SetSupportsMultichannelContent(true, out _);
+		}
+
+		audioSession.SetCategory(AVAudioSessionCategory.PlayAndRecord, AVAudioSessionCategoryOptions.DefaultToSpeaker);
```
Should we set it once or each time before starting the recognition?

brminnick replied:
Ah - yea, good point, we should be setting it every time speech recognition is triggered. I can update it in the morning.

brminnick replied:
Thanks again, Vlad! Updated ✅
